core/App/contexts/auth.tsx

```diff
@@ -30,16 +33,38 @@ export const AuthProvider: React.FC = ({ children }) => {
     await storeWalletSecret(secret)
   }
 
+  const commitPIN = async (useBiometry: boolean): Promise<boolean> => {
+    const secret = walletSecret
+    if (!secret) {
+      return false
+    }
+    if (useBiometry) {
+      await storeWalletSecret(secret, useBiometry)
+    } else {
+      // erase wallet key if biometrics is disabled
+      await wipeWalletKey(useBiometry)
+    }
+    return true
+  }
+
   const checkPIN = async (pin: string): Promise<boolean> => {
-    const secret = await loadWalletSecret()
+    const secret = await loadWalletSecret('', '', false)
```

Comment:
What's the purpose of the boolean?

Reply:
I added that so it would skip trying to load the wallet secret when the user would enter the pin. Before the change if the user declined biometrics and instead used their pin then bcwallet would still try to load the wallet secret from RSA storage. This caused the user to be prompted for biometrics again when entering their pin